lib/fastly_nsq/http/nsqlookupd.rb

```diff
@@ -0,0 +1,99 @@
+# frozen_string_literal: true
+
+require 'fastly_nsq/http'
+
+class FastlyNsq::Http
+  class Nsqlookupd
+    extend Forwardable
+    def_delegator :client, :get
+
+    BASE_NSQLOOKUPD_URL = "http://#{ENV.fetch('NSQLOOKUPD_HTTP_ADDRESS', '').split(',')[0]}".freeze
+
+    ##
+    # List of producers for a given topic
+    #
+    # @param topic [String] the topic for which to list producers
+    def self.lookup(topic:, **args)
+      new(request_uri: '/lookup', **args).get(topic: topic)
+    end
+
+    ##
+    # List of all known topics
+    def self.topics(**args)
+      new(request_uri: '/topics', **args).get
+    end
+
+    ##
+    # List of channels for a given topic
+    #
+    # @param topic [String] the topic for which to list channels
+    def self.channels(topic:, **args)
+      new(request_uri: '/channels', **args).get(topic: topic)
+    end
+
+    ##
+    # List all known nsqd nodes
+    def self.nodes(**args)
+      new(request_uri: '/nodes', **args).get
+    end
+
+    ##
+    # Deletes an existing topic
+    #
+    # @param topic [String] the exsiting topic to delete
+    def self.delete_topic(topic:, **args)
+      new(request_uri: '/delete_topic', **args).get(topic: topic)
+    end
+
+    ##
+    # Deletes an existing channel of an existing topic
+    #
+    # @param topic [String] an exsiting topic
+    # @param channel [String] the exsiting channel to delete
+    def self.delete_channel(topic:, channel:, **args)
+      new(request_uri: '/delete_channel', **args).get(topic: topic, channel: channel)
```

Comment:
This appears to not match the documentation: http://nsq.io/components/nsqd.html#post-channeldelete.